net: lib: tls_credentials: clarify secure tag usage for TLS credentials

mburumaxwell opened the pull request:
Updated documentation to clarify that while credential pairs should generally share the same secure tag for subsystems supporting multiple credentials per tag, some implementations may expect only one credential per tag.

Ref: https://github.com/zephyrproject-rtos/zephyr/pull/88653#discussion_r2162043050

rlubos commented:
Looks good, but the commit msg needs some tweaking, signoff entry is missing and there's a char/line limit exceeded.

jukkar commented:
Note that the signed-off-by needs to be a real id, not a github generated pseudo id. It also needs to match the git commit author id.

mburumaxwell committed:
net: lib: tls_credentials: clarify secure tag usage for TLS credentials

Updated documentation to clarify that while credential pairs should
generally share the same secure tag for subsystems supporting multiple
credentials per tag, some implementations may expect only one credential
per tag.

Signed-off-by: Maxwell Weru <mburumaxwell@gmail.com>
